config/crds/cluster_v1alpha1_machinedeployment.yaml

```diff
@@ -45,9 +45,7 @@ spec:
                 rollingUpdate:
                   properties:
                     maxSurge:
-                      type: object
```

Comment:
Using `type: Object` here  makes the validation always fail, because the underlying type in the struct is `intstr.IntOrString` which means it will be _either_ a string or an int but not an object that contains fields for both